editor/src/viewport_tools/tools/select.rs

```diff
@@ -440,6 +683,49 @@ impl Fsm for SelectToolFsmState {
 					data.drag_current = mouse_position + closest_move;
 					Dragging
 				}
+				(ResizingBounds, MouseMove { .. }) => {
+					if let Some(bounds) = &mut data.bounding_box_overlays {
+						if let Some(movement) = &mut bounds.selected_edges {
+							let mouse_position = input.mouse.position;
+
+							let snapped_mouse_position = data.snap_handler.snap_position(responses, input.viewport_bounds.size(), document, mouse_position);
+
+							let delta = movement.pos_to_scale_transform(snapped_mouse_position);
+
+							let selected = data.layers_dragging.iter().collect::<Vec<_>>();
+							let mut selected = Selected::new(&mut bounds.origional_transforms, &mut bounds.pivot, &selected, responses, &document.graphene_document);
+
+							selected.update_transforms(delta);
+						}
+					}
+					ResizingBounds
+				}
+				(RotatingBounds, MouseMove { snap_angle, .. }) => {
+					if let Some(bounds) = &mut data.bounding_box_overlays {
+						let angle = {
+							let start_vec = data.drag_start - bounds.pivot;
+							let end_vec = input.mouse.position - bounds.pivot;
```

Comment:
Nitpick: using the full word `vector` would help avoid confusion with Rust's `Vec` data type.

Reply:
I think end_offset is probably a better name for this. I have renamed all the other occurrences of this pattern too.

Reply:
Perfect, even better.